client/packages/system/src/Patient/VaccinationCards/VaccineCardTable.tsx

```diff
@@ -0,0 +1,148 @@
+  DataTable,
+  useColumns,
+  NothingHere,
+  useFormatDateTime,
+  useRowStyle,
+  useTheme,
+  StatusCell,
+} from '@openmsupply-client/common';
+import { usePatientVaccineCard } from '../api/hooks/usePatientVaccineCard';
+import { VaccinationCardItemFragment } from '../api/operations.generated';
+
+interface VaccinationCardProps {
+  programEnrolmentId: string;
+  openModal: (
+    vaccinationId: string | null | undefined,
+    vaccineCourseDoseId: string
+  ) => void;
+  encounterId?: string;
+}
+
+const useStyleRowsByStatus = (rows?: VaccinationCardItemFragment[]) => {
+  const { setRowStyles } = useRowStyle();
+  const theme = useTheme();
+
+  useEffect(() => {
+    if (!rows) return;
+
+    const doneRows = rows.filter(row => row.given).map(row => row.id);
+    const notDoneRows = rows.filter(row => !row.given).map(row => row.id);
+
+    setRowStyles(doneRows, {
+      backgroundColor: `${theme.palette.background.success} !important`,
+    });
+    setRowStyles(
+      notDoneRows,
+      {
+        backgroundColor: 'white !important',
+      },
+      // Parameter to prevent the previous setRowStyles from being
+      // reset/overwritten
+      false
+    );
+  }, [rows]);
+};
+
+export const VaccinationCardComponent: FC<VaccinationCardProps> = ({
+  programEnrolmentId,
+  openModal,
+  encounterId,
+}) => {
+  const t = useTranslation('dispensary');
+  const { localisedDate } = useFormatDateTime();
+  const theme = useTheme();
+
+  const {
+    query: { data, isLoading },
+  } = usePatientVaccineCard(programEnrolmentId);
+
+  useStyleRowsByStatus(data?.items);
+
+  const columns = useColumns<VaccinationCardItemFragment>([
+    {
+      key: 'age',
+      label: 'label.age',
+      sortable: false,
+      accessor: ({ rowData }) =>
+        t('label.age-months-count', { count: rowData.minAgeMonths }),
+    },
+    {
+      key: 'label',
+      label: 'label.dose',
+      accessor: ({ rowData }) => rowData.label,
+    },
+    {
+      key: 'status',
+      label: 'label.status',
+      accessor: ({ rowData }) => rowData.given,
+      Cell: ({ ...props }) => (
+        <StatusCell
+          {...props}
+          statusMap={{
+            true: {
+              color: theme.palette.success.light,
+              label: t('label.status-given'),
+            },
+            false: {
+              color: theme.palette.error.main,
+              label: t('label.status-not-given'),
+            },
+          }}
+        />
+      ),
+    },
+    {
+      key: 'suggestedDate',
+      label: 'label.suggested-date',
+      accessor: ({ rowData }) => localisedDate(rowData.suggestedDate ?? ''),
+    },
+    {
+      key: 'dateGiven',
+      label: 'label.date-given',
+      accessor: ({ rowData }) => localisedDate(rowData.vaccinationDate ?? ''),
+    },
+  ]);
+
+  const isPreviousDoseGiven = (row: VaccinationCardItemFragment) => {
+    if (!data?.items) return false;
+    const doseIndex = data?.items.findIndex(dose => dose.id === row.id);
```

Comment:
Ohh, actually, this first needs to filter by vaccineCourseId - this only works for the first row in the card 😁 

The vaccine card rows have vaccine_course_id on them, just needs to be exposed to frontend!

Reply:
Ha, good catch! b41fe97cef